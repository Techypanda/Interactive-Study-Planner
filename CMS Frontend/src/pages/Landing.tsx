import { Box, Typography } from "@material-ui/core";
import { DefaultProps } from "../types";

export default function Landing(props: DefaultProps) {

  return (
    <Box id="landing">
      <Typography>Steak baby yeah steak</Typography>
    </Box>
  )
}