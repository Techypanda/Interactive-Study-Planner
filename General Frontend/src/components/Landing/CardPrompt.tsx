import { Box, Paper, Typography } from "@material-ui/core";
import styled from "styled-components";
import { CardPromptProps } from "../../types";

function CardPrompt(props: CardPromptProps) {
  return (
    <div className={props.className}>
      <Box display="flex" justifyContent="center">
        <Paper className="max-width cardprompt" onClick={props.onClick}>
          <Box p={3}>
            <Box mb={2}>
              <Typography variant="h6" component="h6">{props.title}</Typography>
            </Box>
            <img src={props.src} alt={props.alt} className="responsive-img" />
            <Typography variant="body1" align="center">
              {props.description}
            </Typography>
          </Box>
        </Paper>
      </Box>
    </div>
  )
}
export default styled(CardPrompt)`
.max-width {
  max-width: 450px;
}
.cardprompt {
  cursor: pointer;
}
.cardprompt:hover {
  transform: scale(1.1);
}
.responsive-img {
  width: 100%;
  object-fit: contain;
  max-height: 200px;
}
`;