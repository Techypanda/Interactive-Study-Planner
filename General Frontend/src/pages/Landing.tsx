import {  Card, CardHeader, CardMedia, CardContent, Grid, Paper, Typography } from "@material-ui/core";
import Navbar from "../components/shared/Navbar";
import styled from 'styled-components';
import {useHistory} from "react-router-dom";

// pages
import ViewAllCareers from "./ViewAllCareers";
import ViewAllUnits from "./ViewAllUnits";
import PlannerInitialPage from "./PlannerInitialPage";

// stock images
import MedicalImage from "../static/doctor_teacher.jpg";
import CareersImage from "../static/career.jpg";
import TraitsImage from "../static/traits_image.jpg";
import '../App.scss';



function Landing() {
    const history = useHistory();
    return(
	<>
	    <Navbar/>
	    <br/>
	    <br/>
	    <br/>
	    <Grid container spacing={8} justify="center">
		<Grid item xs={12} sm={6}>
		    <Typography variant="h5">
			Hi, welcome to Curtin University's Medical Course Planner!
		    </Typography>
		</Grid>
	    </Grid>
	    <br/>
	    <br/>
	    <br/>
	    <Typography variant="h5" className="classes.prompt">
		What would you like to do?
	    </Typography>
	    
	    {/* routed cards */}
	    <Grid container spacing={4} justify="center">
		<Grid item xs={12} sm={6}>
		    <div className="card-body">
			<Card variant="outlined" onClick = {() => history.push('./PlannerInitialPage') }> 
			    <CardHeader title="Plan Your Medical Course"/>
			    <Grid item>
				<img  src={MedicalImage} alt="medical stock image"/>
			    </Grid>
			    <CardContent>
				<Typography variant="body1" align="center">
				    See how Curtin's flexible course structure can be shaped to you advantage.
				</Typography> 
			    </CardContent>
			</Card>
		    </div>
		</Grid>

		<Grid item xs={12} sm={6}>
		    <div className="card-body">
			<Card variant="outlined"  onClick = {() => history.push('/ViewAllCareers')}>
			    <CardHeader title="See Careers"/>
			    <Grid item>
				<img src={CareersImage} alt="careers stock image"/>
			    </Grid>
			    <CardContent>
				<Typography variant="body1" align="center" >
				    See all the possible careers Curtin can lead to and their requirements.
				</Typography>
			    </CardContent>
			</Card>
		    </div>
		</Grid>

		<Grid item xs={12} sm={6} >
		    <div className="card-body">
			<Card variant="outlined"  onClick = {() => history.push('/ViewAllUnits')}>
			    <CardHeader title="See Classes"/>
			    <Grid item>
				<img src={TraitsImage} alt="careers stock image"/>
			    </Grid>
			    <CardContent>
				<Typography  variant="body1" align="center">
				    Discover all the courses that Curtin University offers in this program.
				</Typography>
			    </CardContent>
			</Card>
		    </div>
		</Grid>
	    </Grid>
	</>
    );
}


export default Landing;
