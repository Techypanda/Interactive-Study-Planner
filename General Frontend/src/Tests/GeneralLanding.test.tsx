import { render } from '@testing-library/react';
import Landing from '../pages/Landing';

describe('Landing General Frontend is working which happens When', () => {
    it('Renders Without A Crash', () => {
        render(<Landing />)
    })
    it('Renders And Matches Snapshot', () => {
        const x = render(<Landing />)
        expect(x).toMatchSnapshot();
    })
})