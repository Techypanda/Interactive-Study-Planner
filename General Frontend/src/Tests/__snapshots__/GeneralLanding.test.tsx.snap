// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing General Frontend is working which happens When Renders And Matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiBox-root MuiBox-root-15"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5"
          >
            Hi, Welcome to Curtin University's Medical Course Planner
          </h2>
        </div>
        <div
          class="MuiBox-root MuiBox-root-16"
        >
          <h3
            class="MuiTypography-root classes.prompt MuiTypography-subtitle1"
          >
            What Would You Like To Do?
          </h3>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-content-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
          >
            <div
              class="MuiBox-root MuiBox-root-17"
            >
              <div
                class="sc-bdfBQB jpocOe"
              >
                <div
                  class="MuiBox-root MuiBox-root-18"
                >
                  <div
                    class="MuiPaper-root max-width cardprompt MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-19"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-20"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-h6"
                        >
                          Plan My Medical Course
                        </h6>
                      </div>
                      <img
                        alt="A Doctor Pointing At A Whiteboard"
                        class="responsive-img"
                        src="doctor_teacher.webp"
                      />
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
                      >
                        See how Curtin's flexible course structure can be shaped to you advantage.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
          >
            <div
              class="MuiBox-root MuiBox-root-21"
            >
              <div
                class="sc-bdfBQB jpocOe"
              >
                <div
                  class="MuiBox-root MuiBox-root-22"
                >
                  <div
                    class="MuiPaper-root max-width cardprompt MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-23"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-24"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-h6"
                        >
                          See Available Careers
                        </h6>
                      </div>
                      <img
                        alt="A Stock Image Declaring Keywords Around Dream Jobs"
                        class="responsive-img"
                        src="career.webp"
                      />
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
                      >
                        See all the possible careers Curtin can lead to and their requirements.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
          >
            <div
              class="MuiBox-root MuiBox-root-25"
            >
              <div
                class="sc-bdfBQB jpocOe"
              >
                <div
                  class="MuiBox-root MuiBox-root-26"
                >
                  <div
                    class="MuiPaper-root max-width cardprompt MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-27"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-28"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-h6"
                        >
                          See Available Units
                        </h6>
                      </div>
                      <img
                        alt="A Stock Image Declaring Units"
                        class="responsive-img"
                        src="traits_image.webp"
                      />
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
                      >
                        Discover all the courses that Curtin University offers in this program.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiBox-root MuiBox-root-15"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5"
        >
          Hi, Welcome to Curtin University's Medical Course Planner
        </h2>
      </div>
      <div
        class="MuiBox-root MuiBox-root-16"
      >
        <h3
          class="MuiTypography-root classes.prompt MuiTypography-subtitle1"
        >
          What Would You Like To Do?
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="MuiBox-root MuiBox-root-17"
          >
            <div
              class="sc-bdfBQB jpocOe"
            >
              <div
                class="MuiBox-root MuiBox-root-18"
              >
                <div
                  class="MuiPaper-root max-width cardprompt MuiPaper-elevation1 MuiPaper-rounded"
                >
                  <div
                    class="MuiBox-root MuiBox-root-19"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-20"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-h6"
                      >
                        Plan My Medical Course
                      </h6>
                    </div>
                    <img
                      alt="A Doctor Pointing At A Whiteboard"
                      class="responsive-img"
                      src="doctor_teacher.webp"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
                    >
                      See how Curtin's flexible course structure can be shaped to you advantage.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="MuiBox-root MuiBox-root-21"
          >
            <div
              class="sc-bdfBQB jpocOe"
            >
              <div
                class="MuiBox-root MuiBox-root-22"
              >
                <div
                  class="MuiPaper-root max-width cardprompt MuiPaper-elevation1 MuiPaper-rounded"
                >
                  <div
                    class="MuiBox-root MuiBox-root-23"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-24"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-h6"
                      >
                        See Available Careers
                      </h6>
                    </div>
                    <img
                      alt="A Stock Image Declaring Keywords Around Dream Jobs"
                      class="responsive-img"
                      src="career.webp"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
                    >
                      See all the possible careers Curtin can lead to and their requirements.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="MuiBox-root MuiBox-root-25"
          >
            <div
              class="sc-bdfBQB jpocOe"
            >
              <div
                class="MuiBox-root MuiBox-root-26"
              >
                <div
                  class="MuiPaper-root max-width cardprompt MuiPaper-elevation1 MuiPaper-rounded"
                >
                  <div
                    class="MuiBox-root MuiBox-root-27"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-28"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-h6"
                      >
                        See Available Units
                      </h6>
                    </div>
                    <img
                      alt="A Stock Image Declaring Units"
                      class="responsive-img"
                      src="traits_image.webp"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
                    >
                      Discover all the courses that Curtin University offers in this program.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
